Update codeowners

AlexGhiondea opened the pull request.

AlexGhiondea committed:
Add codeowners to the file

AlexGhiondea committed:
Add more changes to the codeowners file.

AlexGhiondea committed:
Re-arrange codeowners file

AlexGhiondea committed:
Move the Management and EngSys sections to the top of the file.

weshaggard commented on .github/CODEOWNERS:
```diff
@@ -8,99 +8,760 @@
-# Core
+# >>>>>>>>> Core Libraries <<<<<<<<<<
+
+# > Library: Azure.Core
```
Is this format used for something? Or is it just for documentation?

AlexGhiondea replied:
This is just a visual separator. I can remove if it causes issues or creates confusion!

AlexGhiondea replied:
Changed to `#` 🙂.

weshaggard replied:
I think it might be worth removing as it might cause confusion. Without knowing all the details of the structure of this file I might expect this is required and maps to something somewhere when it is really just documentation, but is mostly duplicated information with other things so perhaps not worth having.

weshaggard commented on .github/CODEOWNERS:
```diff
@@ -8,99 +8,760 @@
 /sdk/                                           @AlexGhiondea
 
-# Core
+# >>>>>>>>> Core Libraries <<<<<<<<<<
```
I must admit that when I say this I thought it was a merge conflict. You might want to reconsider using the ">>>" format.

AlexGhiondea replied:
Ah! YES! Good point 🙂. I'll change that! :)

AlexGhiondea replied:
Done

AlexGhiondea committed:
Remove angled brackets from the file to not make it look like we have merge conflicts.

meeraharidasa commented on .github/CODEOWNERS:
```diff
@@ -1,106 +1,767 @@
+
+# > Service: AKS
+# ServiceLabel: %AKS %Service Attention
+#/<NotInRepo>/            @Azure/aks-pm
```
sdk/aks/

meeraharidasa committed:
Update CODEOWNERS

added some more folders to service labels I found

AlexGhiondea commented on .github/CODEOWNERS:
```diff
@@ -78,7 +78,7 @@
 # > Service: ARM - Core
 # ServiceLabel: %ARM - Core %Service Attention
-#/<NotInRepo>/            @armleads-azure
+/sdk/core/Azure.Core           @armleads-azure
```
Azure.Core is not the right one for this one.

meeraharidasa replied:
Ah okay my bad - we can leave it as NotInRepo instead then

AlexGhiondea commented on .github/CODEOWNERS:
```diff
@@ -296,7 +296,7 @@
 # > Service: Cosmos
 # ServiceLabel: %Cosmos %Service Attention
-#/<NotInRepo>/            @Wmengmsft @MehaKaushik @shurd @anfeldma-ms @zfoster @kushagraThapar @southpolesteve
+/sdk/cosmosdb/          @Wmengmsft @MehaKaushik @shurd @anfeldma-ms @zfoster @kushagraThapar @southpolesteve
```
This one might just be for the resource manager for Cosmos... not for all of Cosmos... We should ask

AlexGhiondea commented on .github/CODEOWNERS:
```diff
@@ -528,7 +528,7 @@
 # > Service: Network
 # ServiceLabel: %Network %Service Attention
-#/<NotInRepo>/            @aznetsuppgithub
+sdk/network/           @aznetsuppgithub
```
This folder only contains the resource manager for network... That might be the right thing -- but not 100% sure.

AlexGhiondea committed:
Update CODEOWNERS

Update 2 owners.

AlexGhiondea committed:
Update CODEOWNERS

Update the ARM - Core entry to not point to Azure.Core

AlexGhiondea committed:
Remove the extra comment around the actual entries

Doing this so that we reduce the amount of comments in the file which could lead to confusion.

weshaggard commented on .github/CODEOWNERS:
```diff
@@ -1,106 +1,586 @@
+# ServiceLabel: %App Services %Service Attention
+#/<NotInRepo>/            @antcp @AzureAppServiceCLI
 
+# ServiceLabel: %Attestation %Service Attention
```
I'd consider removing `%Service Attention` duplication in this file and have the tool that sets up the rule add this label automatically.

AlexGhiondea replied:
Talked about this offline -- we'll leave this in for now.

weshaggard reviewed (approved):
Made one comment but otherwise looks good.

AlexGhiondea committed:
Update CODEOWNERS

Add the Event Hubs, Event Grid and Service Bus owners.

AlexGhiondea committed:
Update CODEOWNERS

Update Event Hubs CODEOWNER

jsquire commented:
@AlexGhiondea:  You may want to consider pulling in the additions from #16200

jsquire commented on .github/CODEOWNERS:
```diff
@@ -1,106 +1,586 @@
+
+# ######## Management Plane ########
+
+/**/*Management*/                               @allenjzhang @m-nash @markcowl @YalinLi0312 @bquantump @nisha-bhatia
```
This one is broad and catches things like changes to `sdk/servicebus/Azure.Messaging.ServiceBus/tests/Administration/ServiceBusManagementClientLiveTests.cs`.   I don't have a good suggestion for tuning it, but maybe you do?

AlexGhiondea replied:
I will leave that as-is for now... this particular one is only used for the PR owner suggestions.

jsquire commented on .github/CODEOWNERS:
```diff
@@ -1,106 +1,586 @@
+# ServiceLabel: %Data Lake %Service Attention
+#/<NotInRepo>/            @sumantmehtams
+
+# ServiceLabel: %Data Lake Storage Gen1 %Service Attention
```
Is the distinction between Gen1 and Gen2 germane to routing?  As triage, I'm often unclear on the generation and they route to the same contacts.   Maybe consider removing these and collapsing into  the "Data Lake Store" label instead?

AlexGhiondea replied:
No -- the gen1 is a thing about Data Lake.... They have 2 gens.

jsquire replied:
I know they have two generations.... but they both end up going to the same team.   Making the distinction during triage is often a "well, Gen2 because.... reasons!" decision.    Figured it may be worth collapsing.

jsquire commented on .github/CODEOWNERS:
```diff
@@ -1,106 +1,586 @@
+# ServiceLabel: %Event Grid %Service Attention
 # PRLabel: %Event Grid
-/sdk/eventgrid/                                 @Kishp01 @ahamad-MS
+/sdk/eventgrid/                                 @Kishp01 @ahamad-MS @jfggdl
```
Do we want Josh Love on this list as well?   What about Matt Ellis?

ahamad-MS replied:
Please add Josh Love for .NET SDK

AlexGhiondea replied:
Done.

jsquire commented on .github/CODEOWNERS:
```diff
@@ -1,106 +1,586 @@
+# ServiceLabel: %IotDPS %Service Attention
+#/<NotInRepo>/            @iluican @jlian
+
+# ServiceLabel: %IotHub %Service Attention
```
There's also a track two for the various  IoT bits under development at `sdk/iot/` - should we include that here?

AlexGhiondea replied:
I think we should -- do you know which ones to add here?

jsquire replied:
I do not, but @drwill-ms may have some insight to offer.

drwill-ms replied:
I think the track1 control-plane SDKs have been owned by @iluican. We shouldn't have started track2 control-plan yet AFAIK.

We have an effort for hub data-plane track2 SDK. That would be my team of usual suspects:  @drwill-ms @timtay-microsoft @abhipsaMisra @vinagesh @azabbasi @bikamani @barustum @jamdavi

jsquire commented on .github/CODEOWNERS:
```diff
@@ -1,106 +1,586 @@
+/sdk/search/Microsoft.*/                        @brjohnstmsft @arv100kri @bleroy @tjacobhi @markheff @miwelsh
+
+# PRLabel: %Service Bus (Client)
 /sdk/servicebus/                                @JoshLove-msft @ShivangiReja @jsquire @MiYanni
```
Does Shivangi still want to be spammed with SB stuff, or should we trim the list?

AlexGhiondea replied:
I think we should remove her, but I would really like to get this list trimmed.

jsquire commented on .github/CODEOWNERS:
```diff
@@ -1,106 +1,586 @@
+# ServiceLabel: %Search %Service Attention
+/sdk/search/Microsoft.*/                        @brjohnstmsft @arv100kri @bleroy @tjacobhi @markheff @miwelsh
+
+# PRLabel: %Service Bus (Client)
```
Is that `(Client)` bit a typo or will that assign the client label?    I could have overlooked something, but I don't see you using that syntax elsewhere.

AlexGhiondea replied:
Ouch -- this is a good point! I need to fix this.

AlexGhiondea replied:
Fixed.

kasobol-msft commented on .github/CODEOWNERS:
```diff
@@ -1,106 +1,586 @@
 
-# PRLabel: %Tables
-/sdk/tables/                                    @christothes
+# ServiceLabel: %Storage %Service Attention
+/sdk/storage/Azure.Storage.*/                   @amishra-dev @seanmcc-msft @amnguye @kasobol-msft @tg-msft @xgithubtriage
```
what is xgithubtriage going to do with storage PRs?

btw. please make sure this https://github.com/Azure/azure-sdk-for-net/pull/16200 gets reflected here correctly.

kasobol-msft replied:
https://github.com/Azure/azure-sdk-for-net/pull/16363 as well.

AlexGhiondea replied:
@kasobol-msft what should we have instead of xgithubtriage? This was in the internal wiki page.

AlexGhiondea replied:
@kasobol-msft I pulled in those 2 PRs.

AlexGhiondea commented:
/check-enforcer override

jsquire commented:
@AlexGhiondea:  You'll likely want to absorb the additions from ##16440 as well.

AlexGhiondea committed:
Update CODEOWNERS

Update EventGrid.

AlexGhiondea committed:
Update CODEOWNERS

Fix issue with the PR Label.

AlexGhiondea committed:
Update CODEOWNERS

Incorporate #16200

AlexGhiondea committed:
Update CODEOWNERS

Incorporate #16363

AlexGhiondea committed:
Update CODEOWNERS

Incorporate #16440

AlexGhiondea committed:
Merge branch 'master' into UpdateCodeowners

AlexGhiondea commented:
/check-enforcer override

AlexGhiondea committed:
Update CODEOWNERS

Ensure that for WebJobs we label PRs.

AlexGhiondea committed:
Update CODEOWNERS

Clarify the storage section.

drwill-ms commented on .github/CODEOWNERS:
```diff
@@ -1,107 +1,596 @@
 # PRLabel: %DigitalTwins
-/sdk/digitaltwins/                              @drwill-ms @timtay-microsoft @abhipsaMisra @vinagesh @azabbasi @bikamani @barustum @jamdavi
+# ServiceLabel: %Digital Twins %Service Attention
+/sdk/digitaltwins/                              @drwill-ms @timtay-microsoft @abhipsaMisra @vinagesh @azabbasi @bikamani @barustum @sourabhguha @inesk-vt @jamdavi
```
Why did @sourabhguha and @inesk-vt get added? I'm not against it, but noticing this is a change. They may not want all the PR email.

AlexGhiondea replied:
This comes from the internal wiki list. I will add them there for now and if there is too much traffic we can remove.

drwill-ms replied:
I checked with Saurabh, and he asked me to remove him. I'm checking with Ines.

Is there any benefit to them to be on this list, aside from PR participation?

drwill-ms commented on .github/CODEOWNERS:
```diff
@@ -1,107 +1,596 @@
+#/<NotInRepo>/            @madhurinms
 
 # PRLabel: %Iot
 /sdk/iot/                                       @drwill-ms @timtay-microsoft @abhipsaMisra @vinagesh @azabbasi @prmathur-microsoft @bikamani @barustum @jamdavi
```
you can remove @prmathur-microsoft on this, if you like. She has moved on to another team.

AlexGhiondea replied:
Done.

AlexGhiondea committed:
Update CODEOWNERS

Remove person not on the team anymore.

AlexGhiondea commented:
/check-enforcer override
